packages/modeling/src/measurements/measureCenterOfMass.js

```diff
@@ -0,0 +1,90 @@
+const flatten = require('../utils/flatten')
+
+const vec3 = require('../maths/vec3')
+
+const geom2 = require('../geometries/geom2')
+const geom3 = require('../geometries/geom3')
+const path2 = require('../geometries/path2')
+
+const measureBoundingBox = require('./measureBoundingBox')
+
+const cacheOfCenterOfMass = new WeakMap()
+
+/*
+ * Measure the center of mass for the given geometry.
+ * The given geometry is infinitely thin, so there's no real mass, just a center.
```

Comment:
I'm a little concerned that this becomes misleading.  If there's no mass, as asserts, there's no center of mass, and it should return null, or throw an error.

However, if you consider the center of mass to be what the center would be for, say, a piece of paper, then this would return an incorrect result.  

Either way, this is surprising.

Reply:
Yeah. There's probably a center of mass calculation for 2D solids as well. I'll research. This might be mid-point (average of all points).

Having said that... I think that for a path, there is no 'mass' as there is truly no mass in any direction. The path is infinitely thin, no volume, no width, no height, no depth.

I guess the question is.. what's the default center of mass for non-geometries?

FYI, measure bonding box returns [0,0,0] for non-geometries.

Reply:
Found the formula for the 2D center of mass from Paul Bourke. It's a centroid based on the area of the 2D polygon.

[centroid.pdf](https://github.com/jscad/OpenJSCAD.org/files/6680165/centroid.pdf)

Reply:
@SimonClark please review again.

center of mass calculations for geom2 have changed. this should be a far better implementation.